Stardrop/ViewModels/ModDownloadViewModel.cs

```diff
@@ -0,0 +1,162 @@
+
+        // --Composite or dependent properties--
+
+        private readonly ObservableAsPropertyHelper<double> _completion = null!;
+        public double Completion => _completion.Value;
+
+        private readonly ObservableAsPropertyHelper<bool> _isSizeUnknown = null!;
+        public bool IsSizeUnknown => _isSizeUnknown.Value;
+
+        private readonly ObservableAsPropertyHelper<string> _downloadSpeedLabel = null!;
+        public string DownloadSpeedLabel => _downloadSpeedLabel.Value;
+
+        private readonly ObservableAsPropertyHelper<string> _downloadProgressLabel = null!;
+        public string DownloadProgressLabel => _downloadProgressLabel.Value;
+
+        // --Commands--
+
+        public ReactiveCommand<Unit, Unit> CancelCommand { get; }
+        public ReactiveCommand<Unit, Unit> RemoveCommand { get; }
+
+        public ModDownloadViewModel(Uri modUri, string name, long? sizeInBytes, CancellationTokenSource downloadCancellationSource)
+        {
+            _startTime = DateTimeOffset.UtcNow;
+
+            ModUri = modUri;
+            _name = name;
+            _sizeBytes = sizeInBytes;
+            _downloadedBytes = 0;
+            _downloadCancellationSource = downloadCancellationSource;
+
+            CancelCommand = ReactiveCommand.Create(Cancel);
+            RemoveCommand = ReactiveCommand.Create(Remove);
+
+            // SizeBytes null-ness to IsSizeUnknown converison
+            this.WhenAnyValue(x => x.SizeBytes)
+                .Select(x => x.HasValue is false)
+                .ToProperty(this, x => x.IsSizeUnknown, out _isSizeUnknown);
+
+            // DownloadedBytes to DownloadSpeedLabel conversion
+            this.WhenAnyValue(x => x.DownloadedBytes)
+                .Sample(TimeSpan.FromMilliseconds(500), RxApp.MainThreadScheduler)
+                .Select(bytes =>
+                {
+                    double elapsedSeconds = (DateTimeOffset.UtcNow - _startTime).TotalSeconds;
+                    double bytesPerSecond = bytes / elapsedSeconds;
+                    if (bytesPerSecond > 1024 * 1024)  // MB 
+                    {
+                        return $"{(bytesPerSecond / (1024 * 1024)):N2} MB/s";
+                    }
+                    else if (bytesPerSecond > 1024) // KB
+                    {
+                        return $"{(bytesPerSecond / 1024):N2} KB/s";
+                    }
+                    else // Bytes
+                    {
+                        return $"{bytesPerSecond:N0} B/s";
+                    }
+                }).ToProperty(this, x => x.DownloadSpeedLabel, out _downloadSpeedLabel);
+
+            // DownloadedBytes and SizeBytes to DownloadProgressLabel conversion
+            this.WhenAnyValue(x => x.DownloadedBytes, x => x.SizeBytes)
+                .Sample(TimeSpan.FromMilliseconds(500), RxApp.MainThreadScheduler)
```

Comment:
Having download counters update as fast as they can is surprisingly disconcerting. Once every half-second looks about right.